src/randomx.cpp

```diff
@@ -102,6 +103,12 @@ extern "C" {
 	}
 
 	randomx_dataset *randomx_alloc_dataset(randomx_flags flags) {
+
+		//fail on 32-bit systems if DatasetSize is >= 4 GiB
```

Comment:
Shouldn't the failure threshold be much smaller? At 4GiB there is no room in the address space for any other code or data.

Reply:
If the computer doesn't have enough virtual memory (which is likely) then the allocation will fail naturally.

This condition is in place to prevent an overflow which could make the allocation to falsely succeed. For example if `DatasetSize` is 4.5 GiB, it will overflow to 0.5 GiB on 32-bit systems and only 0.5 GiB will be allocated, leading to runtime crashes.

Reply:
ok